### ISSUE TITLE
Add Merge Account to NewExpensify

### ISSUE THREAD

**Issue by zanyrenney** [Expensify/App#47073]
# [DESIGN DOC ➡️](https://docs.google.com/document/d/1QhfgsJ0w-KoWX7r3RYNSk25DICp4C8XIfrFmULZgEgk/edit)

## Tasks

- [x] Post **Proposal (full Problem/Solution statement)** in `#whatsnext`
- [x] Wait at least one full business day, and until the post has a majority (2/3) of positive reactions (👍)
- [x] Paste Proposal in the space above [with a link to the Slack thread](https://expensify.slack.com/archives/CC7NECV4L/p1724098861385259)
- [x] Email `strategy@expensify.com` and paste in the Proposal
- [x] Host a **pre-design** meeting ([example](https://expensify.slack.com/archives/C01GTK53T8Q/p1665794669539419)) in an appropriate slack channel to discuss any necessary details in public before filling out the **High-level of proposed solution** section.
- [x] Fill out the **High-level overview of the problem**, **Timeline**, **Terminology**, and **High-level of proposed solution** sections of the Design Doc
- [x] Email `strategy@expensify.com` (continue the same email chain as before - your last message should be the WN Proposal) with the link to your Design Doc containing your high-level problem and solution
- [x] Add the `DesignDocReview` label to get the **High-level overview of the problem** and **High-level of proposed solution** section reviewed
- [x] Respond to any questions or concerns and bring up blockers in Slack to get a consensus if necessary
- [x] Confirm that the doc has the minimum necessary number of reviews before proceeding
- [x] Host another pre-design meeting in the appropriate slack channel to ask for engineering feedback on the technical solution.
- [x] Fill out the **Detailed implementation of the solution** and related sections.
- [x] Re-add the `DesignDocReview` label to this issue
- [x] Respond to any questions or concerns and bring up blockers in Slack to get consensus if necessary
- [x] Confirm that the doc has the minimum necessary number of reviews before proceeding
- [x] Email `strategy@expensify.com` one last time to let them know the Design Doc is moving into the implementation phase
- [x] Implement the changes
- [ ] Add regression tests so that QA can test your feature with every deploy ([instructions](https://stackoverflowteams.com/c/expensify/questions/363))
- [ ] Send out a follow up email to `strategy@expensify.com` once everything has been implemented and do a **Project Wrap-Up** retrospective that provides:
    - Summary of what we accomplished with this project
    - What went well?
    - What could we have done better?
    - What did we learn?

<details><summary>Issue Owner</summary>Current Issue Owner: @arosiclair</details>

**Comment by zanyrenney** [comment]
Need to improve this but startign with the P/S 

**Proposal:** add the remaining account settings to new dot, Part 1: Merge Account (Part 2: Device Management TBC)
**Problem:**
When users accidentally create personal accounts while their company is using Expensify, they face several issues. They must either:

- Re-scan their receipts using their company account.
- Manually create expenses within their company account.
- Accept that they won’t be reimbursed by their employer.

Generally, users can’t share reports directly with their company in-app, and managing two accounts can be confusing. Users might also unknowingly create expenses that don't align with company policies.

**Solution:**
Rebuild the ability to "Merge Accounts" in NewExpensify. This feature allows users to easily merge their personal and company accounts on Expensify.

This will improve the transfer from individual expense dat to company-created accounts.

**Comment by zanyrenney** [comment]
Pre-design on merging accounts for workspace feed/card conversation: https://expensify.slack.com/archives/C036QM0SLJK/p1723841346915409

**Comment by zanyrenney** [comment]
This is a bit mean, I haven't even done it yet 🥲

**Comment by zanyrenney** [comment]
https://github.com/search?q=org:Expensify+%22merge+account%22&type=Code

**Comment by zanyrenney** [comment]
https://github.com/search?q=org:Expensify+%22merge+account%22&type=Code
https://github.com/Expensify/Web-Expensify/blob/main/script/bwm/HandleBackgroundTasksOnMergeAccount.php
https://github.com/Expensify/Auth/blob/main/auth/command/MergeAccount.cpp

**Comment by zanyrenney** [comment]
Not sure we need @dannymcclain here yet so will remove this from clogging up his K2.

**Comment by zanyrenney** [comment]
![2024-08-20_17-35-28](https://github.com/user-attachments/assets/24d0fc09-6ea8-4fe0-89b7-d358400f01e2)

**Comment by zanyrenney** [comment]
@twisterdotcom and I considered getting the pre-design out today or tomorrow but I realised we need to email strat@ prior, so done that and then can get the pre-design in flight!

**Comment by twisterdotcom** [comment]
Back from some OOO. @zanyrenney has been predesigning in a canvas in my absence.

**Comment by zanyrenney** [comment]
pre-design is posted! https://expensify.slack.com/archives/C06ML6X0W9L/p1725276863731789

**Comment by zanyrenney** [comment]
Going to get some mock help due to the time pressure on this!

**Comment by zanyrenney** [comment]
reassigning @dannymcclain

**Comment by zanyrenney** [comment]
Figma file for reference: https://www.figma.com/design/bHql8diuUEo4HktqZULhLY/Merge-Accounts?node-id=4-4221&node-type=CANVAS&t=d52rTOi706bLXkWx-0

**Comment by zanyrenney** [comment]
Sync with Danny to go over this today cc @dannymcclain 

Then hoping to share a visual pre-d and then move ahead with HL.

**Comment by zanyrenney** [comment]
**User journeys:**

Success:
LHN > Avatar > Security > Merge Accounts > RHP (same as close account)
![2024-09-09_14-46-30](https://github.com/user-attachments/assets/115ac645-05b7-461f-8696-640306a69e0a)

**Comment by zanyrenney** [comment]
**Failure flow 1:** 

Incorrect Magic Link Code (existing pattern) 
![2024-09-09_14-54-57](https://github.com/user-attachments/assets/5a6bef95-f482-4ea1-bab8-be1abf9073d7)

**Comment by zanyrenney** [comment]
**Failure flow 2:**

SAML is set as `required` on the Domain being merged _into_

LHN > Avatar > Security > Merge Accounts > RHP (same as close account) > Magic Code > Error message. 

"SAML....." [placeholder] need to ask James for copy.

**Comment by zanyrenney** [comment]
- How to handle the aftermath?

**Comment by dannymcclain** [comment]
Yo @zanyrenney I got some stuff [here](https://www.figma.com/design/bHql8diuUEo4HktqZULhLY/Merge-Accounts?node-id=1-2&node-type=CANVAS&t=OkYIHa3EPLAX9VGz-11) for you to check out.

Let me know if you want me to post in the predesign threads in Slack for feedback or if you want to handle that!

For the "confirmation" (it worked or it didn't), I think we can handle it with either an RHP like we do for upgrades, or a confirmation modal which we use for tons of stuff. Not sure which I think is more appropriate, so we may just want to get feedback on that during the predesign.

**Comment by zanyrenney** [comment]
Hey @dannymcclain these are great, thanks so much. I want to add a few other things / Qs to the pre-design so I will post them today and ofc tag you!

**Comment by zanyrenney** [comment]
I got held up on customer hiring / syncs, and wanted to check a few things with Ted too. So I will post these _tomorrow!_

**Comment by zanyrenney** [comment]
conversations ongoing, pre-design was posted here - https://expensify.slack.com/archives/C06ML6X0W9L/p1726066356389699

**Comment by zanyrenney** [comment]
will move to doc soon!

**Comment by zanyrenney** [comment]
### **Updates to mocks 1:**

- [ ] Create the corresponding "Domain admin" view where they are receiving a message via concierge that says "zany@expensify.com has attempted to merge zany@gmail.com into their primary account. As Expensify.com has SAML configured, you'll need to verify this request" Approve | Deny - @dannymcclain 
- [ ] Add concierge message to user with outcome of the attempted merge - @dannymcclain 
- [ ] Success: "Your domain admin approved your merge request, zany@gmail.com has been merged into zany@expensify.com" (question for James: should this say [domainadminsemail] instead of just "domain admin" - user might not know what domain admin is?) 
- [ ] Failure: "Your domain admin has denied your merge request. zany@gmail.com remains a separate Expensify account" (question for James: should this say [domainadminsemail] instead of just "domain admin" - user might not know what domain admin is?)

### **Updates to mocks 2:**

- [ ] Get James to approve the copy on these initial mobile mocks for all the flows

### **Updates to mocks 3:**

- [ ] Ask Danny to please create all the mocks in web as well as mob.

**Comment by zanyrenney** [comment]
Message to domain admin from concierge > message reads [approve/deny merge request] > approve deny pill ?

**Comment by zanyrenney** [comment]
Ask Ted in doc review

- for the flows where we are saying you can't merge into the account, for smart scanner and invoicing, should we have the logout link in the error message?

**Comment by jamesdeanexpensify** [comment]
I am still working through copy suggestions in the mocks [here](https://www.figma.com/design/bHql8diuUEo4HktqZULhLY/Merge-Accounts?node-id=73-10380&node-type=canvas&t=55eLC7GgZsdbsuQe-0). I think we're pretty close but @zanyrenney @dannymcclain feel free to bring anything out that looks weird or you have questions on, into Slack!

**Comment by bondydaa** [comment]
will dive into what's in the doc tomorrow and answer whatever pings are waiting on me

**Comment by jamesdeanexpensify** [comment]
One last comment thread in the Figma mocks for @zanyrenney. Should be good otherwise!

**Comment by zanyrenney** [comment]
Thank you so much @jamesdeanexpensify

**Comment by bondydaa** [comment]
Will dive into the doc today and post any questions I have there.

**Comment by zanyrenney** [comment]
ty @twisterdotcom

**Comment by zanyrenney** [comment]
@bondydaa is diving into the API section today.

**Comment by bondydaa** [comment]
started spelunking in the code and started on the API section. kicked off a question to slack that i'm not sure about but i'll keep plugging away on it to have it ready for monday/tuesday sending.

**Comment by bondydaa** [comment]
discussion ongoing in this thread to make sure I cover/address API/Onyx changes required for the HL, still hoping we can get this sent out for review by EOD today though might end up being tomorrow since Zany may be offline by the time I finish my updates.

**Comment by zanyrenney** [comment]
Removing star from myself - @bondydaa comment above should clear the overdue as the co-lead!

**Comment by bondydaa** [comment]
okay think data storage / api changes is all set for now, at least I've tried to document where I wasn't sure and can hopefully get the feedback from the team then to get to the right spot. I resolved the rest of the outstanding items i had open and left some suggestions for @zanyrenney on that last remaining question you'd tagged me in. 

I think we're in a good spot to send it out once you review those suggestions.

**Comment by zanyrenney** [comment]
woooo!

**Comment by zanyrenney** [comment]
![2024-09-25_17-13-50](https://github.com/user-attachments/assets/87a4a4a2-f509-445e-ac62-cccdb2a7c505)

shipped!

**Comment by trentpetty** [comment]
I have read and reviewed this Design Doc!

**Comment by flodnv** [comment]
I have read and reviewed this Design Doc!

**Comment by CortneyOfstad** [comment]
I have read and reviewed this Design Doc!

**Comment by danielrvidal** [comment]
I have read and reviewed this Design Doc!

**Comment by AndrewGable** [comment]
I have read and reviewed this Design Doc!

**Comment by bondydaa** [comment]
doc is in review and comments are being addressed

**Comment by twisterdotcom** [comment]
Same as Friday. It's in review.

**Comment by isabelastisser** [comment]
I have read and reviewed this Design Doc!

**Comment by dannymcclain** [comment]
Not overdue. Zany and I are working on a few new mocks for the admin messages/actions.

**Comment by bondydaa** [comment]
I believe most comments are addressed, there is one outstanding from Flo that I will address next week but should be looking good to start on detailed section next week once that is done.

**Comment by zanyrenney** [comment]
Nice @bondydaa sounds good. I'm going to amend the mock ups to show the group chat vs concierge chat for approving the merge request for a SAML domain.

**Comment by bondydaa** [comment]
sorry IFR has not been my friend so far this week, going to devote a good chunk of tomorrow to the doc.

**Comment by zanyrenney** [comment]
Hey @bondydaa how did that day go devoted to the doc? Do you think we'll be able to finish it in the next few days?

**Comment by jamesdeanexpensify** [comment]
Starred @bondydaa as the issue owner for now!

**Comment by bondydaa** [comment]
had to switch focus to https://github.com/Expensify/Expensify/issues/419777 but have PR in review there so should be able to get back onto this doc tomorrow 🙏 

I started searching for a c+ to help me with the detailed portion for new dot changes and I plan to spin up a engineering predesign to get through the remaining outstanding comments on the doc and make sure I havent missed something.

**Comment by bondydaa** [comment]
started working on a more technical pre-design that i'll get posted tomorrow to hopefully get the rest of the remaining comments put to be so we can move on to detailed.

**Comment by bondydaa** [comment]
okay think I have the predesign drafted up now. I'll send it out first thing monday so it doesn't get burried over the weekend.

**Comment by zanyrenney** [comment]
Awesome, nice thinking re the timing @bondydaa

**Comment by bondydaa** [comment]
predesigned started here https://expensify.slack.com/archives/C03TQ48KC/p1729525076634409

**Comment by zanyrenney** [comment]
It looks like the pre-design has largely wound down now, when can we expect the doc to be finished and sent out, @bondydaa ?

**Comment by zanyrenney** [comment]
Thanks for the update!

**Comment by bondydaa** [comment]
spun up this issue for sibtain https://github.com/Expensify/App/issues/51822

I got the predesign bits back into the HL and will start moving onto the detailed portion to scope out all of the required backend changes.

One idea that I'll first be looking into was brought up here https://expensify.slack.com/archives/C03TQ48KC/p1729532966998279?thread_ts=1729525103.903289&cid=C03TQ48KC that could make our lives easier in getting the data back out to all the various clients by forcing them to reconnect via some type of trigger based on the `lastUpdateID` in onxy updates.

This would be more ideal than the route of trying to queue up individual updates within onyx because there is just so much data that needs to be shared out. 

There is already a way this works for clients that are actively subscribed to pusher but our need is for clients not active who would be fetching onyxUpdates. 

From ioni these are 2 potential areas to investigate first 
<img width="1717" alt="image" src="https://github.com/user-attachments/assets/83928b0f-5d98-4758-9e39-31133f13bbf2">

**Comment by tgolen** [comment]
I have read and reviewed this Design Doc!

**Comment by zanyrenney** [comment]
Waiting on another engineer to pick this up. It's being called out in #retain but I am going to approach some others via DM EOW if we still don't have anyone volunteer.

**Comment by marcaaron** [comment]
I don't have a critical issue assigned at the moment so I will jump in here and try to pick things up where Bondy left off.

**Comment by marcaaron** [comment]
I have left a bunch of questions on the doc. Maybe it would be better to pull out into a predesign to get faster movement on all of it though. I think this is the main stuff that needs to be resolved:

- [ ] What if any changes are we making to user's data when merging? There was a discussion about Cards. Some things about Group Chats (though I think they already work as intended). We should clarify the scope of the changes and what kind of compatibility we are targeting or if the way things work today is acceptable and this is just mostly a UI migration.

- [ ] I had a few notes about the flows. Most importantly there is some place where we currently requiring a magic code when a domain admin approves a request to merge account for a `samlRequired` user. We would not be requiring it anymore with the current proposal and unsure if this creates a security risk or not (and I assume we added this for a good reason).

- [ ] There's some dangling conversation about how to update all the clients Onyx data when someone merges. Sounds like we did not exactly land on a satisfying or clear proposal for how to handle this yet. I think we should decide as part of the high level?

Once that stuff is resolved I think the Detailed design can get unblocked.

**Comment by marcaaron** [comment]
I posted a few thoughts about the 3rd bullet point there [here](https://expensify.slack.com/archives/C07NZ8B1VTQ/p1732232982627879) in Slack. It sounds like we kind of made some progress on a solution, but not really clear to me what path was decided on and would like to consider an alternative that would allow us to greatly reduce the scope of this project.

**Comment by marcaaron** [comment]
We also talked a bit about the first bullet in [this thread](https://expensify.slack.com/archives/C07HPDRELLD/p1732236988656299?thread_ts=1732140230.737009&cid=C07HPDRELLD) with @blimpich since we thought there might be overlap. I'm kind of wary of making this project a "fix all the things related to merging accounts" and instead would like to clarify what _exactly_ doesn't work about account merging today and be a bit more strategic how we identify and tackle whatever those individual problems might be.

**Comment by marcaaron** [comment]
Conversations ongoing in Slack.

**Comment by zanyrenney** [comment]
I need to get back to the doc and a few questions @marcaaron had, last week was not viable to do this but should have more time this week.

**Comment by marcaaron** [comment]
Still trying to get some more thoughts on:

>What if any changes are we making to user's data when merging?

Then we should be ready to move forward here.

**Comment by zanyrenney** [comment]
Going to bring a conversation to slack about ab few things on this, drafting it up!

**Comment by marcaaron** [comment]
I think we have resolved most of the high level questions. Aiming to have the detailed design part shipped by EOW.

**Comment by marcaaron** [comment]
Made a bunch of edits and pinged @allroundexperts that we should be good to finish the front end sections now.

**Comment by marcaaron** [comment]
We are getting closer here. Still discussing what to do about the SAML required merge request notification edge case in [this thread](https://expensify.slack.com/archives/C07NZ8B1VTQ/p1733335447869119). 

@allroundexperts we might need to rework one or two things depending on the outcome of that conversation.

**Comment by zanyrenney** [comment]
I will update the HL tomorrow too.

**Comment by zanyrenney** [comment]
Spanish translations have been added and Marc was finishing the manual QA.

**Comment by marcaaron** [comment]
Finished the manual QA section and cleaned up the doc. It's out for final review now!

**Comment by aldo-expensify** [comment]
I have read and reviewed this Design Doc!

**Comment by CortneyOfstad** [comment]
I have read and reviewed this Design Doc!

**Comment by flodnv** [comment]
I have read and reviewed this Design Doc!

**Comment by marcaaron** [comment]
Detailed in review now!

@tylerkaraszewski @zsgreenwald @danielrvidal @AndrewGable if there's any chance of finishing your reviews sooner vs later that would be great. It's a pretty short one and I'd like to get all the GitHubs created (and possibly implement a few things) over the holiday. Or at least have things ready to pass off to someone else since I'm going on leave around EOM.

**Comment by danielrvidal** [comment]
I have read and reviewed this Design Doc!

**Comment by zanyrenney** [comment]
🤦🏼 I obviously did not mean to close. reopened and added back to the right section of the board.

**Comment by AndrewGable** [comment]
I have read and reviewed this Design Doc!

**Comment by deetergp** [comment]
I have read and reviewed this Design Doc!

**Comment by marcaaron** [comment]
This is moving into implementation. @zanyrenney Here are the GH issues:

https://github.com/Expensify/Expensify/issues/455378
https://github.com/Expensify/Expensify/issues/455380
https://github.com/Expensify/Expensify/issues/455382
https://github.com/Expensify/Expensify/issues/455383
https://github.com/Expensify/App/issues/54443

**Comment by marcaaron** [comment]
Looking for someone to take this over. I _could_ start on the backend PRs, but unsure if I will be able to finish them so might make sense to hand off at this point.

**Comment by arosiclair** [comment]
> Looking for someone to take this over. I _could_ start on the backend PRs, but unsure if I will be able to finish them so might make sense to hand off at this point.

Hey I finally took a look at the high level. I can take over once you're out!

**Comment by arosiclair** [comment]
Update thread [here](https://expensify.slack.com/archives/C07NZ8B1VTQ/p1735659034346289)

**Comment by zanyrenney** [comment]
@arosiclair awesome, thank you so much for stepping in here.

**Comment by zanyrenney** [comment]
If there is anything you need help with, let me know!

**Comment by zanyrenney** [comment]
hey @parasharrajat I was chatting to Andrew and it be great to have you aboard. Thank you so much for offering. I've signed you to this issue! We should be able to manage the payment through the typical hub issue creation/PR review process but let me know if you have any questions or concerns!

**Comment by zanyrenney** [comment]
@twisterdotcom @jamesdeanexpensify @dannymcclain do you guys want me to unassign this from you? I don't want to clog your K2 - sorry we should have split out this tracking issue into sub issues for your inputs (copy changes,. design help, pre review for DD) let me know!

**Comment by dannymcclain** [comment]
Works for me!

**Comment by jamesdeanexpensify** [comment]
Sounds good!

**Comment by zanyrenney** [comment]
@arosiclair @parasharrajat Hey team! Would you be able to provide an update with where we are at with this currently, please?

**Comment by arosiclair** [comment]
I added our implementation issues as sub issues ^.

https://github.com/Expensify/Expensify/issues/455383 is the last backend issue to complete. Once that's done https://github.com/Expensify/App/issues/54443 will be unblocked.

**Comment by zanyrenney** [comment]
Thank you - super helpful @arosiclair

**Comment by arosiclair** [comment]
https://github.com/Expensify/Expensify/issues/455383 is in review. Should be done in a day or two.

**Comment by zanyrenney** [comment]
Ongoing!

**Comment by arosiclair** [comment]
Starting work on https://github.com/Expensify/App/issues/54443

**Comment by zanyrenney** [comment]
We're down to the final PR for this now right?

**Comment by arosiclair** [comment]
> We're down to the final PR for this now right?

Yeah that's correct

**Comment by zanyrenney** [comment]
last PR and comment here - https://github.com/Expensify/App/pull/56911#issuecomment-2695343733

**Comment by zanyrenney** [comment]
Hey team, I am ooo @arosiclair please can you keep this udpated in my absence. thanks!

**Comment by arosiclair** [comment]
We're still working on https://github.com/Expensify/App/issues/47073. I believe we're still waiting on feedback from @parasharrajat.

**Comment by zanyrenney** [comment]
@parasharrajat please can you prioritise your feedback? Also going to DM @parasharrajat

**Comment by zanyrenney** [comment]
There are a bunch of bugs that are being found as we continue to develop this. These are being actively worked on though. I have asked when they think it will be ready.

**Comment by zanyrenney** [comment]
I provided an update i retain and bumped the PR.

**Comment by allroundexperts** [comment]
Will raise a PR for these today.

**Comment by arosiclair** [comment]
> Main PR is on staging. QA found out the following issues:

We just put the Merge Accounts feature behind the beta, so we're free to clean these up in follow up PR's. Let's take care polishing these over @allroundexperts @parasharrajat

**Comment by dangrous** [comment]
This one's also related right? https://github.com/Expensify/App/issues/59824

**Comment by allroundexperts** [comment]
@parasharrajat https://github.com/Expensify/App/pull/59911 and https://github.com/Expensify/App/pull/59955 ready for review.

**Comment by zanyrenney** [comment]
@parasharrajat please could you review the issues linked above.

**Comment by arosiclair** [comment]
We're working through a few PRs to cleanup the bugs we found in QA:

- https://github.com/Expensify/App/pull/59911
- https://github.com/Expensify/App/pull/59955
- https://github.com/Expensify/App/pull/60147

These all still need review and testing so they probably won't get merged until next week.

**Comment by allroundexperts** [comment]
Status:

1. https://github.com/Expensify/App/issues/59757  🟢 (https://github.com/Expensify/App/pull/59911)
2. https://github.com/Expensify/App/issues/59759 🟢 (https://github.com/Expensify/App/pull/59911)
3. https://github.com/Expensify/App/issues/59760 🟢 (https://github.com/Expensify/App/pull/59911)
4. https://github.com/Expensify/App/issues/59761  🟢 (https://github.com/Expensify/App/pull/59911)
5. https://github.com/Expensify/App/issues/59763 🟢 (https://github.com/Expensify/App/pull/59955)
6. https://github.com/Expensify/App/issues/59765 🟢 (https://github.com/Expensify/App/pull/59955) 
7. https://github.com/Expensify/App/issues/59767 🟡 (https://github.com/Expensify/App/pull/59911) 
8. https://github.com/Expensify/App/issues/59772 🟡 (https://github.com/Expensify/App/pull/59911)
9. https://github.com/Expensify/App/issues/59779 🔵 
10. https://github.com/Expensify/App/issues/59783 🔵 
11. https://github.com/Expensify/App/issues/59784 🟢 (https://github.com/Expensify/App/pull/59911)
12. https://github.com/Expensify/App/issues/59764 🔴 
13. https://github.com/Expensify/App/issues/59824 🟢 (https://github.com/Expensify/App/pull/59911)
14. https://github.com/Expensify/App/issues/59834 🟢 (https://github.com/Expensify/App/pull/59955)

🟡  -> In progress
🟢  -> Done
🔵 -> Not touched yet
🔴 -> Very trivial. Thinking of ways to solve it.

**Comment by zanyrenney** [comment]
Amazing effort @allroundexperts - thank you so much! For the LHN issue, i see that one is in progress, what kind of ETA are you thinking?

**Comment by zanyrenney** [comment]
How did they reviews go @parasharrajat ?

**Comment by arosiclair** [comment]
Two issues left still:
- https://github.com/Expensify/App/issues/59764 - we're fixing some issues in the PR. Waiting on updates from @allroundexperts 
- https://github.com/Expensify/App/issues/59779 - I identified the likely root cause and need to draft a PR and test it

**Comment by arosiclair** [comment]
https://github.com/Expensify/App/issues/59764 is complete.

I'm working on https://github.com/Expensify/App/issues/59779. ETA is next week.

**Comment by arosiclair** [comment]
https://github.com/Expensify/App/issues/59779 is nearing completion

**Comment by parasharrajat** [comment]
Final PR is merged https://github.com/Expensify/App/pull/60147. 


What is next here @arosiclair? Should this be marked completed after that PR is deployed?

**Comment by arosiclair** [comment]
Once the PR is deployed we're going to do a little bit of testing on staging. If everything looks good, we'll post another PR to remove the beta and close this out.

**Comment by parasharrajat** [comment]
Last PR of this project is now completed. Is there anything pending on this?

**Comment by arosiclair** [comment]
Yes we still have to do this

> Once the PR is deployed we're going to do a little bit of testing on staging. If everything looks good, we'll post another PR to remove the beta and close this out.

**Comment by parasharrajat** [comment]
@allroundexperts Can you please create a draft PR to remove the beta?

**Comment by arosiclair** [comment]
I already drafted the PRs. Please hold

**Comment by zanyrenney** [comment]
### TESTED: - Trying to merge an uncreated account


- [x] Merge accounts appears as menu item
- [x] Is clickable
- [x] RHP appears
- [x] Field appears for entry
- [x] Email address entered
- [x] Blocked as email address doesn't have Expensify account

https://github.com/user-attachments/assets/2348767d-5163-4ac3-bc98-be12aec12377

**Comment by zanyrenney** [comment]
### TESTED: Trying to merge an account with SAML: SAML not rewuired for login: successful


- [x] Merge accounts appears as menu item
- [x] Is clickable
- [x] RHP appears
- [x] Field appears for entry
- [x] Email address entered
- [x] Email appears in mailbox
- [x] Magic code is sent
- [x] Magic code is entered in RHP
- [x] Accounts are successfully merged

https://github.com/user-attachments/assets/877c5cb4-82be-4a1f-9833-c778887c5388

**Comment by arosiclair** [comment]
We're in the process of adding regression tests to QA. Testing special accounts is challenging since Applause does not have easy access to these:

- New SAML accounts
- New SSDos/Expensiworks accounts
- New accounts with invoiced billing enabled

We're discussing that [here](https://expensify.slack.com/archives/C07NZ8B1VTQ/p1750150279702519)

**Comment by parasharrajat** [comment]
What is pending here?

**Comment by zanyrenney** [comment]
Wrap up email has been sent, we can consider this DONE! 🎉 

<img width="1302" height="928" alt="Image" src="https://github.com/user-attachments/assets/5ecf84d6-85f7-4efe-b588-16d0877c43cc" />

**Comment by allroundexperts** [comment]
Can we have a payment summary before closing this @zanyrenney?

**Comment by allroundexperts** [comment]
@zanyrenney In my initial discussion with David Bondy, we decided this to be $3000, and agreed to keep a buffer of $500 if this becomes too complex.

This took a lot of back and forth and I would suggest this to be $3500. This also includes the document writing.

Since @parasharrajat wasn't involved in writing the doc, we can subtract that amount from this for him. I think I was supposed to be paid $1000 for the doc, so we can subtract that.

**Comment by allroundexperts** [comment]
Hey @zanyrenney!

> @allroundexperts we pay the contributor and C+ the same, so there isn't a case of subtracting partial payments from different people here.

Can you please double check this? The design document writing is always separate and the C+ isn't involved in it.

**Comment by arosiclair** [comment]
In this case, we're just assessing the project as a whole. As mentioned, the contributor and C+ usually get paid an equal amount and we agreed $3000 is fair for the both of you.
